### PR TITLE
Allow user to customise definition of acceptable response

### PR DESCRIPTION
**Pull request by alexmojaki**
New take on https://github.com/yola/demands/pull/43
Based on https://github.com/yola/demands/compare/test-dont-review-this?expand=1
@RayeN

### CODE REVIEW

File: demands/__init__.py [hunk 1]
```diff
@@ -131,11 +134,8 @@ def pre_send(self, request_params):
 
     def post_send(self, response, **kwargs):
         """Override this method to modify returned response"""
-        self._demand_success(response, kwargs)
         return response
 
-    def _demand_success(self, response, request_params):
+    def is_acceptable(self, response, **request_params):
```
Comment (RayeN) [inline demands/__init__.py]: just `request_params`?

Comment (RayeN) [inline demands/__init__.py]: Also, probably it makes sense to add some docstring saying that it can be overridden (and in what cases)

**Comment by RayeN** [comment]
:+1: Thanks a lot for this pull.  (are we good with tests?)

**Comment by alexmojaki** [comment]
How's that?

File: tests/test_demands.py [hunk 2]
```diff
@@ -69,6 +69,15 @@ def test_minimal_delete_request(self):
         self.request.assert_called_with(
             method='DELETE', url='http://service.com/delete-endpoint')
 
+    def test_unacceptable_response(self):
+        def get():
+            self.service.get('/get-endpoint')
```
Comment (RayeN) [inline tests/test_demands.py]: I would use `self.service.get('/anypath')` instead of defining own method (save 2 lines :))

**Comment by RayeN** [comment]
:+1:  Looks ok.

**Pull request merged**